ee/clickhouse/views/experiments.py

```diff
@@ -23,6 +27,47 @@
     ProjectMembershipNecessaryPermissions,
     TeamMemberAccessPermission,
 )
+from posthog.utils import generate_cache_key, get_safe_cache
+
+EXPERIMENT_RESULTS_CACHE_DEFAULT_TTL = 60 * 30  # 30 minutes
+
+
+def _calculate_experiment_results(experiment: Experiment):
+    filter = Filter(experiment.filters)
+
+    cache_key = generate_cache_key(f"experiment_{filter.toJSON()}_{experiment.team.pk}")
```

Comment:
Hmm, we need the experiment ID in here too? Can have two different experiments with the same filters, but different experiments

Mind adding tests for this too? 😬

Reply:
do we need the experiment ID? given the same exact filters we should be ok to pull whatever we have cached?

and sure can add tests

Reply:
Let me check, I think the flag key is not in these filters, so results will be for a different experiment 🤔

Reply:
Yep, those filters are too basic, easily can be the same between experiments, it's a combination of these + the flag definition + trend/funnel class which generates the output

Reply:
And dates aren't in the filter either, but directly on the experiment

Reply:
Ok cool, will make sure to get the ID in here!